src/Features/Core/Portable/ExternalAccess/VSTypeScript/VSTypeScriptFormattingService.cs

```diff
@@ -22,8 +22,8 @@ internal sealed class VSTypeScriptFormattingService : IFormattingService
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public VSTypeScriptFormattingService(IVSTypeScriptFormattingServiceImplementation impl)
-            => _impl = impl;
+        public VSTypeScriptFormattingService([Import(AllowDefault = true)] IVSTypeScriptFormattingServiceImplementation impl)
+            => _impl = impl ?? throw new ArgumentNullException(nameof(impl));
```

Comment:
This probably deserves a comment here explaining why we are doing AllowDefault but then still blocking it. Feel free to add as a follow-up since CI is green.